### ISSUE TITLE
Composer run-script break output from behat and atoum

### ISSUE THREAD

**Issue by tyx** [composer/composer#6055]
**Composer version 1.3.1 2017-01-07 18:08:51** 
My `composer.json`:

```json
{
"scripts": {
        "post-install-cmd": [
            "./bin/console cache:clear -vvv"
        ],
        "post-update-cmd": [
            "./bin/console cache:clear -vvv"
        ],
        "ci": [
            "@cs",
            "@analyzer",
            "@tests",
            "@doc"
        ],
        "cs": [
            "php-cs-fixer fix --config=.php_cs.dist -v --dry-run --using-cache=no"
        ],
        "tests": [
            "atoum && behat --strict"
        ],
        "analyzer": [
            "phpstan analyse --no-progress -l 4 src"
        ],
        "doc": [
            "jadd generate app/config/routing.yml app/doc/endpoints.md"
        ]
    }
}
```

Output of `composer diagnose`:

```
Checking composer.json: WARNING
No license specified, it is recommended to do so. For closed-source software you may use "proprietary" as license.
require.rezzza/common : unbound version constraints (dev-master) should be avoided
Checking platform settings: OK
Checking git settings: OK
Checking http connectivity to packagist: OK
Checking https connectivity to packagist: OK
Checking github.com rate limit: OK
Checking disk free space: OK
Checking pubkeys:
Tags Public Key Fingerprint: 57815BA2 7E54DC31 7ECC7CC5 573090D0  87719BA6 8F3BB723 4E5D42D0 84A14642
Dev Public Key Fingerprint: 4AC45767 E5EC2265 2F0C1167 CBBB8A2B  0C708369 153E328C AD90147D AFE50952
OK
Checking composer version: OK
```

When I run this command:

```
composer run-script ci
```

I get the following output:

```
=> PHP 7.0.14 (cli) (built: Dec 27 2016 20:28:03) ( ZTS )
=> Copyright (c) 1997-2016 The PHP Group
=> Zend Engine v3.0.0, Copyright (c) 1998-2016 Zend Technologies
=>     with Zend OPcache v7.0.14, Copyright (c) 1999-2016, by Zend Technologies
> Karibbu\PropertyOnBoarding\Tests\Units\Domain\Contact\ContactDetails...
[..__________________________________________________________][0/2]
[S.__________________________________________________________][1/2]
[SS__________________________________________________________][2/2]
=> Test duration: 0.05 second.
=> Memory usage: 4.00 Mb.
> Karibbu\PropertyOnBoarding\Tests\Units\Domain\Contact\EmailAddress...
[..__________________________________________________________][0/2]
[S.__________________________________________________________][1/2]
[SS__________________________________________________________][2/2]
```

And I expected this to happen:

```
> Karibbu\PropertyOnBoarding\Tests\Units\Domain\Contact\ContactDetails...
[SS__________________________________________________________][2/2]
=> Test duration: 0.04 second.
=> Memory usage: 4.00 Mb.
> Karibbu\PropertyOnBoarding\Tests\Units\Domain\Contact\EmailAddress...
[SS__________________________________________________________][2/2]
```

For behat I have no longer color output !

When running `composer run-script ci`
<img width="868" alt="capture d ecran 2017-01-10 a 10 38 45" src="https://cloud.githubusercontent.com/assets/245494/21801239/2389177e-d721-11e6-8b13-4519b2fba5c0.png">

instead of

<img width="878" alt="capture d ecran 2017-01-10 a 10 39 48" src="https://cloud.githubusercontent.com/assets/245494/21801249/2c2e0718-d721-11e6-9e9e-d6e9b886072d.png">

I saw https://github.com/composer/composer/issues/5933 but it seems not fixed completely for all output

**Comment by alcohol** [comment]
Did you pass the `--ansi` flag? Cause most of those applications you run, use symfony/console, and you need to pass `--ansi` explicitly when running them as a subprocess.

**Comment by tyx** [comment]
Yes I tried ! But it did not work.

**Comment by tyx** [comment]
I had a look to the code and I did not find any call to [Process:setTty](https://github.com/symfony/symfony/blob/9e6d6ba3a9f98d6235f8ee24619a95ca2518c229/src/Symfony/Component/Process/Process.php#L991) so the problem should be here ?

But I guess you have good reasons to not use it ? Any chance to active it on external script ?

**Comment by alcohol** [comment]
That will fail on Windows. And since Composer is cross-platform.. you can do the math :-)

**Comment by tyx** [comment]
So no chance to have this feature in composer ? Even if we active it only for real OS ?

**Comment by alcohol** [comment]
No.

Also keep in mind, "scripts" is there for your convenience, it is not an answer to everything. If it does not work for your situation, do not use it. It looks to me like you would be better of just calling those tools directly.

**Comment by tyx** [comment]
Yes, of course we will do that ;)

`scripts` is just very useful as composer is always here. So if we could avoid another tools only to be use as a runner, it would be great.

Tant pis !

**Comment by alcohol** [comment]
I understand. Most of your issues with it are just visual though. For automated tooling pipelines, that is not really important, is it now?

And for manual runs, you could just call them directly :-)

Just my 2 cents.

**Comment by tyx** [comment]
> For automated tooling pipelines, that is not really important

It is not until first failure. Digging into the whole black and white page to get what was wrong is very hard IMO ;)

> you could just call them directly

And force everyone to remember weird options. I prefer to focus on what it really matters.

But yes, I get we cannot agree on all these details ;)

Anyway thanks for your support !

**Comment by shouze** [comment]
Would like to reopen this issue guys... in fact other tools like yarn/npm do tty detection & have similar scripts feature without this issue, I guess this should be very useful to have it.

**Comment by shouze** [comment]
To be noticed: Windows/tty detection is already done somewhere in the [actual composer source code](https://github.com/composer/composer/blob/4ea017d1ce5046c64edc518a3447f3c1008f4ae1/src/Composer/Command/ConfigCommand.php#L209)

**Comment by shouze** [comment]
Ok so ping @tyx @alcohol seems that have been fixed for behat by default... and for some other tools this can be fixed by specifying following options:
- atoum: --force-terminal
- phpstan, php-cs-fixer & all other tools using symfony console component: --ansi

So.. basicaly in some other echosystems like nodejs one for example this tty detection is done automatically in almost every tool in a better way, each tool detect at runtime, eliminating the need to specify options, would be cool to achieve the same goal.

For behat I need to investigate further to understand from which release this has been fixed but from what I've seen quickly looks like they force coloring output whatever we have a tty or no since some time.

**Comment by zavodyanka** [comment]
Use flag ` --colors`
